gadget/validate.go

```diff
@@ -216,6 +219,15 @@ func validateVolumeContentsPresence(gadgetSnapRootDir string, vol *LaidOutVolume
 			continue
 		}
 		for _, c := range s.Content {
+			if strings.HasPrefix(c.UnresolvedSource, "$kernel:") {
+				// This only validates that the ref is valid.
+				// Resolving happens with ResolveContentPaths()
+				match := assetsKernelRefRE.FindStringSubmatch(c.UnresolvedSource)
```

Comment:
Had a quick look at some validation of socket path and layouts and I'm wondering if we need a regex, or maybe something as simple as:

```
p := c.UnresolvedSource[len("$kernel:"):]
likelyOk :=  filepath.Clean(p) == p &&  !strings.Contains(p, "..") &&  !filepath.IsAbs(p)
```
and then let the filesystem handle the rest.

Reply:
Nvm, looked at #9777 totally forgot the format is slightly different. Maybe regex is actually ok.